<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Nick Youngblut" />

<meta name="date" content="2018-01-10" />

<title>heavy-SIP</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">heavy-SIP</h1>
<h4 class="author"><em>Nick Youngblut</em></h4>
<h4 class="date"><em>2018-01-10</em></h4>


<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a><ul>
<li><a href="#heavy-sip-method-workflow">heavy-SIP method workflow:</a></li>
</ul></li>
<li><a href="#initialization">Initialization</a></li>
<li><a href="#unreplicated-dataset">Unreplicated dataset</a><ul>
<li><a href="#parsing-the-dataset">Parsing the dataset</a></li>
</ul></li>
<li><a href="#one-treatment-control-comparison">One treatment-control comparison</a></li>
<li><a href="#replicated-dataset">Replicated dataset</a><ul>
<li><a href="#t-tests">t-tests</a></li>
<li><a href="#mann-whitney-u-test">Mann Whitney U test</a></li>
</ul></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</div>

<hr />
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="heavy-sip-method-workflow" class="section level2">
<h2>heavy-SIP method workflow:</h2>
<p>Prior to the development of these HTS-SIP analysis methods, DNA- and RNA-SIP experiments that utilized Sanger or high throughput sequencing were usually analyzed with standard statistical processes (e.g. t-tests), in order to identify incorporators. Previous work suggests that these methods generally have low senstivity and/or high false positive rates when applied to sequence data. Here, these analysis methods will be referred to “heavy-SIP” methods. While the work of Youngblut et al., (<a href="https://doi.org/10.1101/138719" class="uri">https://doi.org/10.1101/138719</a>) suggests that HR-SIP analysis methods (eg., MW-HR-SIP) should be used for processing HTS-SIP datasets, the HTSSIP R package provides heavy-SIP methods so researchers have the option of using these methods and making their own comparisons to HR-SIP methods.</p>
<p>heavy-SIP is performed with the <code>heavy_SIP()</code> function, which consists of multiple possible tests. See <code>?heavy_SIP</code> for more details. This vignette demonstrates the use of <code>heavy_SIP()</code>.</p>
</div>
</div>
<div id="initialization" class="section level1">
<h1>Initialization</h1>
<p>First, let’s load some packages including <code>HTSSIP</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dplyr)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(HTSSIP)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># adjusted P-value cutoff </span>
padj_cutoff =<span class="st"> </span><span class="fl">0.1</span></code></pre></div>
</div>
<div id="unreplicated-dataset" class="section level1">
<h1>Unreplicated dataset</h1>
<p>For unreplicated datasets (no experiment replicates of controls or treatments), the options are limited on how to identify incorporators.</p>
<div id="parsing-the-dataset" class="section level2">
<h2>Parsing the dataset</h2>
<p>We will be using a dataset that is already parsed. See <a href="HTSSIP_intro.html">HTSSIP introduction vignette</a> for a description on why dataset parsing (all treatment-control comparisons) is needed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">physeq_S2D2_l</code></pre></div>
<pre><code>## $`(Substrate=='12C-Con' &amp; Day=='3') | (Substrate=='13C-Cel' &amp; Day == '3')`
## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 10001 taxa and 46 samples ]
## sample_data() Sample Data:       [ 46 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 10001 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 10001 tips and 10000 internal nodes ]
## 
## $`(Substrate=='12C-Con' &amp; Day=='14') | (Substrate=='13C-Cel' &amp; Day == '14')`
## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 10001 taxa and 46 samples ]
## sample_data() Sample Data:       [ 46 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 10001 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 10001 tips and 10000 internal nodes ]
## 
## $`(Substrate=='12C-Con' &amp; Day=='14') | (Substrate=='13C-Glu' &amp; Day == '14')`
## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 10001 taxa and 47 samples ]
## sample_data() Sample Data:       [ 47 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 10001 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 10001 tips and 10000 internal nodes ]
## 
## $`(Substrate=='12C-Con' &amp; Day=='3') | (Substrate=='13C-Glu' &amp; Day == '3')`
## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 10001 taxa and 46 samples ]
## sample_data() Sample Data:       [ 46 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 10001 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 10001 tips and 10000 internal nodes ]</code></pre>
</div>
</div>
<div id="one-treatment-control-comparison" class="section level1">
<h1>One treatment-control comparison</h1>
<p>First, we’ll just focus on 1 treatment-control comparison. Let’s get the individual phyloseq object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">physeq =<span class="st"> </span>physeq_S2D2_l[[<span class="dv">1</span>]]
physeq</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 10001 taxa and 46 samples ]
## sample_data() Sample Data:       [ 46 samples by 17 sample variables ]
## tax_table()   Taxonomy Table:    [ 10001 taxa by 8 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 10001 tips and 10000 internal nodes ]</code></pre>
<p>Let’s check that the samples belong to either a 13C-treatment or 12C-control.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">physeq %&gt;%<span class="st"> </span>sample_data %&gt;%<span class="st"> </span>.$Substrate %&gt;%<span class="st"> </span>table</code></pre></div>
<pre><code>## .
## 12C-Con 13C-Cel 
##      23      23</code></pre>
<p>Since this dataset is an unreplicated comparison between treatment &amp; control, we are just going to use the ‘binary’ method, which will call incorporators if they are present in the “heavy” gradient fractions of the treatment and not present in the “heavy” fractions of the control. Note that the “heavy” fractions are user-defined.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res =<span class="st"> </span><span class="kw">heavy_SIP</span>(physeq, <span class="dt">ex=</span><span class="st">&quot;Substrate=='12C-Con'&quot;</span>, 
                   <span class="dt">comparison=</span><span class="st">'H-v-H'</span>, <span class="dt">hypo_test=</span><span class="st">'binary'</span>)
df_res %&gt;%<span class="st"> </span><span class="kw">head</span>(<span class="dt">n=</span><span class="dv">3</span>)</code></pre></div>
<pre><code>##          statistic  p padj
## OTU.514          0 NA   NA
## OTU.816          0 NA   NA
## OTU.1099         0 NA   NA</code></pre>
<p>Since no real statistical test, the “statistic” is just 0 (not an incorporator) or 1 (an incorporator). Also, the “p” and “padj” columns are thus “NA”.</p>
<p>How many “incorporators”?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res$statistic %&gt;%<span class="st"> </span>table</code></pre></div>
<pre><code>## .
##    0    1 
## 3301  957</code></pre>
</div>
<div id="replicated-dataset" class="section level1">
<h1>Replicated dataset</h1>
<p>Experimental replicates allows us to use tradional hypothesis testing (<em>e.g.,</em> t-tests) for determining significantly differ OTU abundances between treatment and controls. Note that there is a reason why more suffisticated statistical methods have been developed for assessing differentially abundant features in high throughput sequencing datasets (<em>e.g.,</em> DESeq2, EdgeR, or MetagenomeSeq). The traditional methods don’t account for many challenging aspects of identifying statistically different abundances in sequence data such as i) a high number of multiple hypotheses ii) zero-inflation iii) compositional data (relative abundances; the sum-to-one constraint).</p>
<p>With that said, let’s try out these heavy-SIP methods on a replicated dataset, with 3 experimental replicates of the control and treatment (total gradients = 6)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">physeq_rep3</code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 6 taxa and 144 samples ]
## sample_data() Sample Data:       [ 144 samples by 5 sample variables ]</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">physeq_rep3 %&gt;%<span class="st"> </span>sample_data %&gt;%<span class="st"> </span><span class="kw">head</span>(<span class="dt">n=</span><span class="dv">3</span>)</code></pre></div>
<pre><code>##                             Gradient Buoyant_density Fraction Treatment
## 12C-Con_rep1_1.668185_1 12C-Con_rep1        1.668185        1   12C-Con
## 12C-Con_rep1_1.680254_2 12C-Con_rep1        1.680254        2   12C-Con
## 12C-Con_rep1_1.679431_3 12C-Con_rep1        1.679431        3   12C-Con
##                         Replicate
## 12C-Con_rep1_1.668185_1         1
## 12C-Con_rep1_1.680254_2         1
## 12C-Con_rep1_1.679431_3         1</code></pre>
<div id="t-tests" class="section level2">
<h2>t-tests</h2>
<p>To compare “heavy” fractions in the treatment versus “heavy” fractions in the control, we will use the “H-v-H” comparison method. See <code>?heavy_SIP</code> for details on other possible comparisons.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res =<span class="st"> </span><span class="kw">heavy_SIP</span>(physeq_rep3, <span class="dt">ex=</span><span class="st">&quot;Treatment=='12C-Con'&quot;</span>, 
                   <span class="dt">comparison=</span><span class="st">'H-v-H'</span>, <span class="dt">hypo_test=</span><span class="st">'t-test'</span>)
df_res %&gt;%<span class="st"> </span><span class="kw">head</span>(<span class="dt">n=</span><span class="dv">3</span>)</code></pre></div>
<pre><code>##         statistic         p      padj
## OTU.1 -0.09563691 0.5336654 0.5641843
## OTU.2 -0.18450596 0.5641843 0.5641843
## OTU.3  1.14859160 0.2280138 0.3519149</code></pre>
<p>“padj” is p-values adjusted with the Benjamini Hochberg method.</p>
<p>How many incorporators?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res %&gt;%
<span class="st">  </span><span class="kw">filter</span>(padj &lt;<span class="st"> </span>padj_cutoff) %&gt;%
<span class="st">  </span>nrow</code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>No incorporators. Obviously, the sensitivity of this method is pretty low. What’s the distribution of p-values?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res$p %&gt;%<span class="st"> </span>summary</code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1619  0.2180  0.2313  0.3228  0.4589  0.5642</code></pre>
</div>
<div id="mann-whitney-u-test" class="section level2">
<h2>Mann Whitney U test</h2>
<p>Does anything change when we use a nonparametric test? Here, we will use the Mann Whitney U test (a nonparametric t-test).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res =<span class="st"> </span><span class="kw">heavy_SIP</span>(physeq_rep3, <span class="dt">ex=</span><span class="st">&quot;Treatment=='12C-Con'&quot;</span>, 
                   <span class="dt">comparison=</span><span class="st">'H-v-H'</span>, <span class="dt">hypo_test=</span><span class="st">'wilcox'</span>)
df_res %&gt;%<span class="st"> </span><span class="kw">head</span>(<span class="dt">n=</span><span class="dv">3</span>)</code></pre></div>
<pre><code>##       statistic         p      padj
## OTU.1         2 0.6666667 0.6666667
## OTU.2         2 0.6666667 0.6666667
## OTU.3         4 0.1666667 0.2500000</code></pre>
<p>What’s the p-value and adjusted-pvalue distribution?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res$p %&gt;%<span class="st"> </span>summary %&gt;%<span class="st"> </span>print</code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1667  0.1667  0.1667  0.3333  0.5417  0.6667</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df_res$padj %&gt;%<span class="st"> </span>summary %&gt;%<span class="st"> </span>print</code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.2500  0.2500  0.2500  0.3889  0.5625  0.6667</code></pre>
<p>Again, no incorporators. The change in abundances must be pretty dramatic for heavy-SIP methods to ID incorporators, especially when there’s many multiple hypotheses.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1>Session info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.3 (2017-11-30)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.3 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
## LAPACK: /usr/lib/atlas-base/atlas/liblapack.so.3.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=C              
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bindrcpp_0.2    phyloseq_1.20.0 HTSSIP_1.3.1    ggplot2_2.2.1  
## [5] tidyr_0.7.2     dplyr_0.7.4    
## 
## loaded via a namespace (and not attached):
##   [1] nlme_3.1-131               bitops_1.0-6              
##   [3] matrixStats_0.52.2         bit64_0.9-7               
##   [5] doParallel_1.0.11          RColorBrewer_1.1-2        
##   [7] rprojroot_1.2              GenomeInfoDb_1.12.3       
##   [9] tools_3.4.3                backports_1.1.1           
##  [11] R6_2.2.2                   coenocliner_0.2-2         
##  [13] vegan_2.4-4                rpart_4.1-11              
##  [15] Hmisc_4.0-3                DBI_0.7                   
##  [17] lazyeval_0.2.0             BiocGenerics_0.22.1       
##  [19] mgcv_1.8-22                colorspace_1.3-2          
##  [21] permute_0.9-4              ade4_1.7-8                
##  [23] nnet_7.3-12                tidyselect_0.2.2          
##  [25] gridExtra_2.3              DESeq2_1.16.1             
##  [27] bit_1.1-12                 compiler_3.4.3            
##  [29] Biobase_2.36.2             htmlTable_1.9             
##  [31] DelayedArray_0.2.7         labeling_0.3              
##  [33] scales_0.5.0               checkmate_1.8.5           
##  [35] genefilter_1.58.1          stringr_1.2.0             
##  [37] digest_0.6.12              foreign_0.8-69            
##  [39] rmarkdown_1.6              XVector_0.16.0            
##  [41] base64enc_0.1-3            pkgconfig_2.0.1           
##  [43] htmltools_0.3.6            htmlwidgets_0.9           
##  [45] rlang_0.1.2                RSQLite_2.0               
##  [47] bindr_0.1                  jsonlite_1.5              
##  [49] BiocParallel_1.10.1        acepack_1.4.1             
##  [51] RCurl_1.95-4.8             magrittr_1.5              
##  [53] GenomeInfoDbData_0.99.0    Formula_1.2-2             
##  [55] biomformat_1.4.0           Matrix_1.2-11             
##  [57] Rcpp_0.12.13               munsell_0.4.3             
##  [59] S4Vectors_0.14.7           ape_4.1                   
##  [61] stringi_1.1.5              yaml_2.1.14               
##  [63] MASS_7.3-47                SummarizedExperiment_1.6.5
##  [65] zlibbioc_1.22.0            rhdf5_2.20.0              
##  [67] plyr_1.8.4                 blob_1.1.0                
##  [69] grid_3.4.3                 parallel_3.4.3            
##  [71] lattice_0.20-35            Biostrings_2.44.2         
##  [73] splines_3.4.3              multtest_2.32.0           
##  [75] annotate_1.54.0            locfit_1.5-9.1            
##  [77] knitr_1.17                 igraph_1.1.2              
##  [79] GenomicRanges_1.28.6       geneplotter_1.54.0        
##  [81] reshape2_1.4.2             codetools_0.2-15          
##  [83] stats4_3.4.3               XML_3.98-1.9              
##  [85] glue_1.1.1                 evaluate_0.10.1           
##  [87] latticeExtra_0.6-28        data.table_1.10.4-2       
##  [89] foreach_1.4.3              gtable_0.2.0              
##  [91] purrr_0.2.4                assertthat_0.2.0          
##  [93] xtable_1.8-2               survival_2.41-3           
##  [95] tibble_1.3.4               iterators_1.0.8           
##  [97] memoise_1.1.0              AnnotationDbi_1.38.2      
##  [99] IRanges_2.10.5             cluster_2.0.6</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
